### PR TITLE
Validation and Model spec

### PR DESCRIPTION
**Pull request by muhammedb8826**
### In this Milestone I have worked on the followings:
- Add the following validations:

    - For the User model:
        Name must not be blank.
        PostsCounter must be an integer greater than or equal to zero. 
     - For the Post model:
         Title must not be blank.
         Title must not exceed 250 characters.
         CommentsCounter must be an integer greater than or equal to zero.
         LikesCounter must be an integer greater than or equal to zero.

- Added unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  subject { Post.new(author_id: 1, title: 'Prayer', comments_counter: 0, likes_counter: 0) }
+
+  before { subject.save }
```
Comment (TadesseDev) [inline spec/post_spec.rb:6]: - [ ] You did a great job on the tests @muhammedb8826, but according to the requirements, we need to test all the custom methods in every model. Hence we need to test the `update_post_counter` and `recent_comments` methods inside the `Post` model.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @muhammedb8826 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/post_spec.rb (hunk 1 above)
Comment (muhammedb8826) [reply spec/post_spec.rb:6]: Thank you for the feedback I will do that

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @muhammedb8826 , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Bravo, testing the methods pointed out by the previous reviewer. ✔
- You successfully completed all the requirements.  🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**